inbm-lib/inbm_lib/trtl.py

```diff
@@ -193,7 +193,7 @@ def image_pull_public(self, image: str, reference: Optional[str], file_name: Opt
     @staticmethod
     def _send_password(cmd: str, password: str) -> Tuple[str, Optional[str], int]:
 
-        p = Popen(shlex.split(str(cmd)), stdout=PIPE, stdin=PIPE, stderr=PIPE)
+        p = Popen(shlex.split(str(cmd)), shell=False, stdout=PIPE, stdin=PIPE, stderr=PIPE)
         pwd = bytes(password + '\n', 'utf-8')
         (out, err) = p.communicate(input=pwd)
```

Comment:
**<h2>$\color{#15F4EE}Miscellaneous$</h2>**
**Suggestion:** Use the 'with' statement to ensure that resources are properly managed and the Popen process is terminated correctly.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
        with Popen(shlex.split(str(cmd)), shell=False, stdout=PIPE, stdin=PIPE, stderr=PIPE) as p:
            pwd = bytes(password + '\n', 'utf-8')
            (out, err) = p.communicate(input=pwd)
```